src/ray/raylet/worker.h

```diff
@@ -221,7 +221,7 @@ class Worker : public WorkerInterface {
 
   bool IsAvailableForScheduling() const {
     return !IsDead()                        // Not dead
-           && !GetAssignedTaskId().IsNil()  // No assigned task
+           && !GetAssignedTaskId().IsNil()  // Task won't cause dead lock, since progress is being made.
```

Comment:
hmm, what about idle worker that doesn't have assigned task or actor? cc @rkooo567

Reply:
The naming `IsAvailableForScheduling` is also a bit confusing.

According to its only usage at `node_manager.cc`, I guess `WontCauseResourceDeadlock` would be a better name.